src/components/MAPI/clusters/ClusterDetail/ClusterDetailOverview.tsx

```diff
@@ -97,6 +98,60 @@ const ClusterDetailOverview: React.FC<{}> = () => {
     [orgId, clusterId]
   );
 
+  const labels = useMemo(() => getVisibleLabels(cluster), [cluster]);
+
+  const [labelsError, setLabelsError] = useState('');
+  const [labelsIsLoading, setLabelsIsLoading] = useState(false);
+
+  const handleLabelsChange = async (patch: ILabelChange) => {
```

Comment:
Once this file gets larger I will probably extract all this logic into custom React hooks, but not worrying about that right now